extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchHolderObjectStrategy.java

```diff
@@ -67,4 +74,32 @@ public HllSketchHolder fromByteBufferSafe(ByteBuffer buffer, int numBytes)
         )
     );
   }
+
+  private boolean isSafeToConvertToNullSketch(ByteBuffer buf)
+  {
+    // TODO: Might need a sanity check here, to ensure that position and offset makes sense.
```

Comment:
The contract is that it should be safe to read `size` bytes from the `ByteBuffer`, so just pass size in here and trust that you can read that many bytes.

Reply:
Added a check here